mrmustard/lab_dev/circuit_components.py

```diff
@@ -317,6 +319,60 @@ def from_quadrature(
         BBBB = QtoB_ib @ (QtoB_ik @ QQQQ @ QtoB_ok) @ QtoB_ob
         return cls._from_attributes(BBBB.representation, wires, name)
 
+    def to_quadrature(self, phi: float = 0.0) -> CircuitComponent:
+        r"""
+        Returns a circuit component with the quadrature representation of this component
+        in terms of A,b,c.
+
+        Args:
+            phi: The quadrature angle. ``phi=0`` corresponds to the x quadrature,
+                    ``phi=pi/2`` to the p quadrature. The default value is ``0``.
+        Returns:
+            A circuit component with the given quadrature representation.
+        """
+        if isinstance(self.representation, Fock):
+            raise NotImplementedError("Not implemented with Fock representation.")
+
+        from .circuit_components_utils.b_to_q import BtoQ
+
+        BtoQ_ob = BtoQ(self.wires.output.bra.modes, phi).adjoint
+        BtoQ_ib = BtoQ(self.wires.input.bra.modes, phi).adjoint.dual
+        BtoQ_ok = BtoQ(self.wires.output.ket.modes, phi)
+        BtoQ_ik = BtoQ(self.wires.input.ket.modes, phi).dual
+        QQQQ = BtoQ_ib @ (BtoQ_ik @ self @ BtoQ_ok) @ BtoQ_ob
+        return QQQQ
+
+    def quadrature_triple(self, phi: float = 0.0) -> tuple | ComplexTensor:
+        r"""
+        The quadrature representation triple A,b,c of this circuit component.
+        """
+        if isinstance(self.representation, Fock):
+            raise NotImplementedError("Not implemented with Fock representation.")
+
+        QQQQ = self.to_quadrature(phi=phi)
+        return QQQQ.representation.data
+
+    def quadrature(self, quad: Batch[Vector], phi: float = 0.0) -> tuple | ComplexTensor:
+        r"""
+        The (discretized) quadrature basis representation of the circuit component.
+        """
+
+        if isinstance(self.representation, Fock):
+            fock_arrays = self.representation.array
+            # Find where all the bras and kets are so they can be conjugated appropriately
+            conjugates = [False] * len(self.wires.indices)
+            conjugates = [
+                conjugates[i] if i in self.wires.ket.indices else True
+                for i in range(len(self.wires.indices))
+            ]
+            quad_basis = math.sum(
+                [quadrature_basis(array, quad, conjugates, phi) for array in fock_arrays], axes=[0]
+            )
+            return quad_basis
+
+        QQQQ = self.to_quadrature(phi=phi)
+        return QQQQ.representation.ansatz(quad)
```

Comment:
Should be able to do 
```suggestion
        return QQQQ.representation(quad)
```